src/VecSim/vec_sim_tiered_index.h

```diff
@@ -53,6 +53,10 @@ class VecSimTieredIndex : public VecSimIndexInterface {
         this->SubmitJobsToQueue(this->jobQueue, this->jobQueueCtx, jobs.data(), callbacks.data(),
                                 jobs.size());
     }
+    template <typename... Args>
+    void log(const char *level, const char *msg, Args... args) const {
+        this->backendIndex->log(level, msg, args...);
+    }
```

Comment:
So for each call to log, we will have an additional function symbol in the binary file 
for example: ("%s, %s", "meow", "charlie") and ("%s",meow) yields 2 symbols
I suggest that the tiered index will hold a copy to the backend index log function